### PR TITLE
Ladybird: Implement an Inspector window for the AppKit chrome

### PR DESCRIPTION
**Pull request by trflynn89**
This implementation contains the DOM tree and CSS style table (computed properties, resolved properties, and variables). You can click nodes in the inspector's DOM tree to further inspect them, or right-click on elements on the page to inspect them from the context menu.

Screenshot 2023-09-13 at 10 48 02 PM

Screenshot 2023-09-13 at 10 47 40 PM

### CODE REVIEW

File: Ladybird/AppKit/UI/Inspector.mm [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+- (NSInteger)outlineView:(NSOutlineView*)view
+    numberOfChildrenOfItem:(id)item
+{
+    if (item == nil) {
+        item = self.dom_tree;
+    }
+
+    NSArray* children = [item objectForKey:@"children"];
+    if (children == nil)
+        return 0;
```
Comment (bugaevc) [inline Ladybird/AppKit/UI/Inspector.mm:384]: You don't have to do these `nil` checks, this is Objective-C 😀

Comment (trflynn89) [reply Ladybird/AppKit/UI/Inspector.mm]: Good point!

**Review by ADKaster — COMMENTED** [review]
tfw you never hit comment to finish your review

File: Ladybird/AppKit/Utilities/NSString+Ladybird.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#import <System/Cocoa.h>
+
+@interface NSString (Ladybird)
+
+- (NSString*)stringByCollapsingConsecutiveWhitespace;
```
[the remark this answers is not in the document]
Comment (bugaevc) [reply Ladybird/AppKit/Utilities/NSString+Ladybird.h:13]: It is a common convention

**Pull request merged**